### ISSUE TITLE
CA1822 fires on a get-only property that has an attribute on the getter

### ISSUE THREAD

**Issue by dbremner** [dotnet/roslyn-analyzers#2834]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.8 (Latest)

#### Diagnostic ID

[CA1822](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1822-mark-members-as-static)

#### Repro steps

1. Create a new console application in .NET Framework or .NET Core
2. Add the following code to a new file called Example.cs
```
using System.Diagnostics;

namespace ConsoleApp
{
    public sealed class Example
    {
        public Example(string value) => Value = value;

        public string Value
        {
            [DebuggerStepThrough]
            get;
        }

        public override string ToString() => Value;
    }
}
```

#### Expected behavior
CA1822 should not trigger on Example.Value.

#### Actual behavior
Receive a C1822 diagnostic that Value does not access instance data.

**Comment by dbremner** [comment]
The same warning is shown for the System.Diagnostics.Contracts.PureAttribute.

**Comment by Evangelink** [comment]
The same code without the attribute on top of the getter results in no diagnostic being raised.

**Comment by Evangelink** [comment]
Looks like a combination of unexpected behaviors is leading to this case.

For some reason, there is no block operation callback if the readonly property getter is not marked with an attribute while there is when marked. Personally, I would expect both cases to generate a block callback. 

The other surprising point is that constructors blocks are skipped https://github.com/dotnet/roslyn-analyzers/blob/master/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs#L317
leading to the fact that the code is missing the assignation of the property to a value coming from a parameter.

Lastly, I have some trouble understanding why properties, especially auto-properties, are flagged. It seems quite rare to have `static` properties compared to methods. I could see the point of flagging a property only returning/dealing with a constant/static value but auto-properties kindda mean I want an instance "object".

@mavasani do you have any explanation or opinion on this?

**Comment by Evangelink** [comment]
Ping @mavasani

**Comment by Evangelink** [comment]
Ping @mavasani

**Comment by Evangelink** [comment]
@mavasani I think on this case the problem is not only the attribute but mainly that the property is initialized from a value coming from the ctor (not mandatory but often the case with readonly properties) so there is no way to make the property static.

Once again I suggest to have a wider change toward properties rather than trying to handle cases one by one.
